### ISSUE TITLE
Pre-compile mongoose/browser

### ISSUE THREAD

**Issue by vkarpov15** [Automattic/mongoose#7219]
So people don't have to merge their webpack config with [Mongoose's recommended config](https://mongoosejs.com/docs/browser.html#webpack). Re: discussion from https://github.com/Automattic/mongoose/issues/6826#issuecomment-435933307

**Comment by saiichihashimoto** [comment]
Here's a really good guide on how to do this:

https://webpack.js.org/guides/author-libraries/

**Comment by captaincaius** [comment]
So in my monorepo project I just finally got around to setting up a child package that depends on mongoose and builds it with webpack, and I was thinking about PRing something, but I realized my goals probably aren't the same as other people's and this would warrant more discussion first.  I'm not sure how it's possible to satisfy everyone's needs with this.

To illustrate what I mean, I'll paste the webpack config I'm using and why...

```
const paths = require('path');
  
module.exports = {
  entry: require.resolve('mongoose/browser.js'),
  output: {
    filename: 'index.js',
    path: paths.resolve(__dirname, ''),
    libraryTarget: 'commonjs2',
  },
  externals: [
    /^node_modules\/.+$/
  ],
  module: {
    rules: [
      {
        include: [
          /\/mongoose\//i,
          /\/kareem\//i,
        ],
        loader: 'babel-loader',
      }
    ]
  },
  node: {
    dns: 'empty',
    fs: 'empty',
    module: 'empty',
    net: 'empty',
    tls: 'empty',
  },
  target: 'web',
  mode: 'production',
};

```

As you can see, it's not all too different from the webpack test, but:

- I need it to be built so it can be consumed by webpack yet again via angular's CLI and build process.
- I need it to work with SSR, so "window" is a no-no.  For example, the npm package "mongoose-browser" breaks SSR because of how it's presumably built.
- Because of angular's purist approach with the CLI, just importing mongoose/browser doesn't work.

So in my case, webpack needs to do very little.

I have a feeling these goals might even be contradictory to what was in mind for this issue.  Unless I'm missing something, I'm not sure how to make everyone happy without actually having at least two different mongoose-browser versions :-/.  Or maybe my goals are just too specific for the scope of this issue.

Even so, there are some questions as to how it should be organized (where it should be output, testing, etc).  I'd be happy to pitch in if you have some guidance for me.

**Comment by captaincaius** [comment]
Since I never heard back about specifics, I decided to just throw together a PR - if any of the details need work I'll just address them thereafter ;).

**Comment by hasezoey** [comment]
should this issue be closed because #8350 is merged?
